### ISSUE TITLE
Issue with ChromaDB Embeddings

### ISSUE THREAD

**Issue by BillJones-SectorFlow** [llmware-ai/llmware#404]
Hi!

Thanks for the awesome project!  Was just looking for some guidance on Chroma embeddings.

I've been using postgres/pg-vector without issue, but wanted to give ChromaDB a shot, but I can't seem to get the embeddings saved without errors.  It looks like embedding_class.create_new_embedding should be a list of embeddings as per chroma's validations, but it is getting a NumPy array?

Any thoughts on what might be wrong?

**Relevant code below:**

```python
libr = Library().load_library(library_name="test_collection", account_name="test_account")

output_results = libr.add_files(input_folder_path="/workspace/tmp")

libr.install_new_embedding(embedding_model_name="BAAI/bge-large-en-v1.5", vector_db="chromadb")
``` 

```
File "/workspace/llmware/llmware/embeddings.py", line 104, in create_new_embedding
    embedding_status = embedding_class.create_new_embedding(doc_ids, batch_size)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/llmware/llmware/embeddings.py", line 2176, in create_new_embedding
    self._collection.add(ids=ids,
  File "/root/miniconda3/envs/secflow/lib/python3.11/site-packages/chromadb/api/models/Collection.py", line 96, in add
    """The total number of embeddings added to the database
                                            ^^^^^^^^^^^^^^^
  File "/root/miniconda3/envs/secflow/lib/python3.11/site-packages/chromadb/api/models/Collection.py", line 349, in _validate_embedding_set
    where=valid_where,
  File "/root/miniconda3/envs/secflow/lib/python3.11/site-packages/chromadb/api/types.py", line 355, in validate_embeddings
    "$in",
ValueError: Expected embeddings to be a list, got [[-6.44383729e-02 -1.44667607e-02  4.17735875e-02 -7.56826177e-02
  -1.43465966e-01 -6.67456770e-04 -1.66972056e-02 .........]]
``` 

**NOTE:** The /workspace/tmp directory just has one of the PDF files from Setup().load_sample_files().

**ALSO NOTE:** I've tried with "mini-lm-sbert" with the exact code from test_embeddings.py and I get the same error.

**For example, I see the same error when massaging the following code into my project:**

```python
model=ModelCatalog().load_model("mini-lm-sbert")
embedding_handler = EmbeddingHandler(library=library)
Status(library.account_name).tail_embedding_status(library.library_name, "mini-lm-sbert")
embedding_summary = embedding_handler.create_new_embedding(embedding_db=embedding_db)
```

**Comment by BillJones-SectorFlow** [comment]
Adding:

```python
if isinstance(vectors, np.ndarray):
    vectors = vectors.tolist()
```
to embeddings.py ~line 2171 seems to make it work, but I'm not sure if that's a proper fix.

**Comment by BillJones-SectorFlow** [comment]
@MacOS Thanks for the info!  After trying to replicate the issue with the suggested example, and being unable to, I eventually tracked the problem down to a conflict in the ChromaDB version seemingly caused by a conflict in version of grpcio needed by pymilvus and the version needed by the latest chromadb.  Since I'm not using milvus, I just forced install of the 0.4.22 version of ChromaDB, ignoring the pymilvus incompatibility and this seemed to have corrected the issue I was having.

Thanks again for looking into it.
